### ISSUE TITLE
Notebook v7 menus should match the v6 ones whenever possible

### ISSUE THREAD

**Issue by fperez** [jupyter/notebook#6398]

**Comment by andrii-i** [comment]
As I mentioned in another thread, we have 2 [release blocking issues](https://github.com/jupyter/notebook/issues?q=is%3Aissue+is%3Aopen+label%3A%22tag%3ARelease+Blocker%22) left and I wanted to align on what has to be done to release Notebook 7. Ease of transition and need to not invalidate Notebook v6 (educational content) are big parts of JEP 79.  Would you say we have achieved sufficient consistency for the release and can address remaining points post-release?

**Comment by andrii-i** [comment]
- Moving to 7.0.x for further follow up after Notebook 7 release. Related comment: https://github.com/jupyter/notebook/issues/6307#issuecomment-1587710396

**Comment by fperez** [comment]
Thanks for checking, @andrii-i! I have to admit I've been busy with other things and haven't looked in a while - I've also fully transitioned to JLab even for all my undergraduate teaching since I last wrote here, so now I'm much less of a reasonable judge :)

I trust the team has done a huge amount of work on this, so we're probably at the point where it's best to make a release, announce it, and plan on responding to the (inevitable) small issues that people always find once a package goes into much wider usage.  But I see the list of known blockers is now down to zero, and I'm not sure waiting much more is beneficial.

There's another reason to release soon, I think: it's early summer, and it's a good time for educators who are some of the big Notebook constituents, to test the materials they may use in the fall semester, still having time for the cycle of feedback and potential changes that may arise from that.

In summary, my 10,000 ft perspective is that we're probably better off moving forward with a release at this point, planning on perhaps a few point updates, so that by August (start of the fall teaching season in many countries), there's a 7.0.x (for a hopefully small value of x) that meets just about everybody's needs.

**Comment by andrii-i** [comment]
Thank you for the feedback @fperez!

**Comment by dbpengra** [comment]
I also support making the menus consistent between the new and old versions of the notebook.  

Another problem I have is that the Help menu in the new version does not provide links to the usual pile of online documentation for Python and all of the major packages (NumPy, Matplotlib, etc.).  These were/are very helpful!

**Comment by goelakash** [comment]
Want to take a stab at this. Rather than a single sweeping audit, I think it makes sense to split it menu by menu:

**A. File menu** — add `Download` and a `Download as` submenu of nbconvert exporters from inside an open notebook. This is the specific gap called out in the original post and still the case today.

**B. View menu** — surface things NB 7 supports but doesn't expose (Toggle Line Numbers being the obvious one), and bring back the Cell Toolbar submenu (None / Edit Metadata / Raw cell format / Slideshow / Attachments / Tags).

**C. Edit menu** — most v6 Edit items already live under Edit/Run in NB 7, so a lot of this is "relocated, not missing." Will post a short diff before coding so we don't argue about it in review.

**D. Help menu** — ~~bring back the external doc links (Python, IPython, Markdown, NumPy, SciPy, Matplotlib, SymPy, pandas), per @dbpengra's earlier comment.~~ Additional issues noted here: https://github.com/jupyter/notebook/issues/6398#issuecomment-4640514744

@jtpio — does this split look reasonable? Planning to start with A since it's the smallest and the most concretely requested. Also curious whether this should stay on the 7.6.0 milestone or slip.

**Comment by andrii-i** [comment]
Hi @goelakash. For what's it's worth, split looks reasonable to me.

I'd say feel free to proceed and open PR for A) while you're waiting for feedback from @jtpio.

**Comment by goelakash** [comment]
Did a side-by-side of File menus between v6 (`notebook==6.5.7`) and v7 (`notebook==7.5.7`).

| v6 File menu enties | v7 File menu entries | Status |
|---|---|---|
| New Notebook ▸ | New ▸ | no change needed (v7 submenu now also holds Console/Terminal/Text File, so v6 label no longer fits the contents) |
| Open… | Open… | no change needed |
| — | New Console for Notebook | no change needed (v7 only, fine) |
| Make a Copy… | Duplicate | could be renamed back to "Make a Copy…" |
| Save as… | Save Notebook As… | could be renamed back to "Save as…" |
| Rename… | Rename… | no change needed |
| Save and Checkpoint (⌘S) | Save Notebook (⌘S) — file save only, no checkpoint | needs addition + maybe further work |
| — | Save All | no change needed (v7 only, fine) |
| — | Reload Notebook from Disk | no change needed (v7 only, fine) |
| Revert to Checkpoint ▸ | Revert Notebook to Checkpoint… | could be renamed back to "Revert to Checkpoint" |
| Print Preview | — | needs addition + maybe further work |
| Download as ▸ | Save and Export Notebook As ▸ | could be renamed back to "Download as" |
| — | Download | no change needed (v7 only, fine) |
| Trust Notebook | Trust Notebook | no change needed |
| Close and Halt | Close and Shut Down Notebook | could be renamed back to "Close and Halt" |

**Comment by krassowski** [comment]
> should be renamed to "Download as"

Many of these changes were intentional and based on multiple cycles of UX research and user feedback. I don't think we use _should_ lightly here. In this example users were surprised by the auto-save happening before downloading a notebook as a PDF which is required for nbconvert to export what users sees. Of course ideally this would not be the case but I am just saying that old is not necessarily better.

**Comment by goelakash** [comment]
Yeah agreed, the language is just an indication, not a recommendation

@krassowski updated

**Comment by goelakash** [comment]
Here's some additional deep-dive on the differences between v6 and v7 notebook Help menu

  | Help / About entry | v6 | v7 | Notes |
  |---|---|---|---|
  | User Interface Tour | yes | — | tracked separately in #6411 |
  | Notebook Help | yes | — | no equivalent |
  | Edit Keyboard Shortcuts | in Help | moved to Settings | relocated |
  | Jupyter Reference | — | yes | v7-only (Lab static link) |
  | JupyterLab FAQ | — | yes | v7-only — JupyterLab branding leak |
  | JupyterLab Reference | — | yes | v7-only — JupyterLab branding leak |
  | Markdown Reference (top group) | — | yes | duplicate of the one at the bottom |
  | About the Python 3 (ipykernel) Kernel | inlined into About | separate dialog | split out of About |
  | About Jupyter | — | yes | v7-only |
  | Documentation | — | yes | v7-only |
  | About → server Python version | shown | missing | regression |
  | About → active kernel banner | shown | moved to separate dialog | regression |

### v6 
##### Help

<img width="188" height="386" alt="Image" src="https://github.com/user-attachments/assets/2c84afd0-c04d-4833-a790-a5db9860945d" />

##### About

<img width="714" height="431" alt="Image" src="https://github.com/user-attachments/assets/29a0f7df-8e40-4b36-9aac-ae68e161e486" />

### v7
##### Help

<img width="354" height="464" alt="Image" src="https://github.com/user-attachments/assets/785a4536-67f4-4661-a9e1-80b82370f54c" />

##### About

<img width="417" height="321" alt="Image" src="https://github.com/user-attachments/assets/7cb88ad9-3024-460d-9948-81ad16768595" />

**Comment by goelakash** [comment]
@jtpio @krassowski quick question - do you think we should verify if the notebook 7.x behaviour aligns with what the menu items suggest (e.g. Save and Export -> should also save before exporting, etc) and that they were not just a rename? That would close the page on this discussion.
